File: data/tojson.jl

using Dates
using JSON

# Read file

f = open("data.txt")

lines = readlines(f)

close(f)

# Collect data into nested dictonary

dictparent = Dict()

global station, dictchild

for line in lines

    if (ismatch(r"^:", line))
        
        station = replace(line, ":" => "")

        dictchild = Dict(
            "time" => [],
            "prec" => [],
            "tair" => [],
            "qpast" => [],
            "qfuture" => []
        )

    else

        values = Meta.parse.(split(line))

        push!(dictchild["time"], DateTime(values[1], values[2], values[3] ,values[4]))
        push!(dictchild["prec"], values[5])
        push!(dictchild["tair"], values[6])
        push!(dictchild["qpast"], values[14])
        push!(dictchild["qfuture"], values[15])
        
        dictparent[station] = dictchild

    end

end

# Write to json

f = open("data.json", "w")

JSON.print(f, dictparent)

close(f)

File: data/tojson.py

import re
import json

with open("data.txt") as f:
    lines = f.readlines()

lines = [line.strip() for line in lines] 

datacomplete = {}

for line in lines:
    if re.match(r':', line):
        station = line.replace(":", "")
        datasingle = {
            "time": [],
            "prec": [],
            "tair": [],
            "qpast": [],
            "qfuture": []
        }
    else:
        values = line.split()
        datasingle["time"].append(values[0] + "-" + values[1] + "-" + values[2] + " " + values[3] + ":00")
        datasingle["prec"].append(float(values[4]))
        datasingle["tair"].append(float(values[5]))
        datasingle["qpast"].append(float(values[13]))
        datasingle["qfuture"].append(float(values[14]))
        datacomplete[station] = datasingle

with open('data.json', 'w') as outfile:
    json.dump(datacomplete, outfile)